test/parameterized_test_test.dart

```diff
@@ -0,0 +1,400 @@
+// ignore_for_file: avoid_print
+
+import 'package:parameterized_test/parameterized_test.dart';
+import 'package:parameterized_test/src/errors/parameterized_error.dart';
+import 'package:parameterized_test/src/parameterized_test.dart';
+import 'package:parameterized_test/src/test_options/test_options.dart';
+import 'package:parameterized_test/src/test_options/value_with_test_options.dart';
+import 'package:test/test.dart';
+
+import 'test_helpers.dart';
+
+void main() {
+  late TestHelper groupMock;
+  late TestHelper testMock;
+  late TestHelper setUpMock;
+  late TestHelper tearDownMock;
+  late ParameterizedTestImpl pTest;
+
+  setUp(() {
+    groupMock = TestHelper();
+    testMock = TestHelper();
+    setUpMock = TestHelper();
+    tearDownMock = TestHelper();
+    pTest = ParameterizedTestImpl(
+      groupMock.test,
+      testMock.test,
+      setUpMock.setUp,
+      tearDownMock.setUp,
+    );
+  });
+
+  group('makeDescription tests', () {
+    const baseDescription = 'Group description';
+    test(
+        'makeDescription return a String with '
+        'comma separated values', () {
+      final values = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+      final result = pTest.makeDescription(baseDescription, 1, values, null);
+
+      expect(result, '[ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]');
+    });
+
+    test(
+        'makeDescription return a String with '
+        'single qouted strings', () {
+      final values = ['one', 'two', 'three'];
+
+      final result = pTest.makeDescription(baseDescription, 1, values, null);
+
+      expect(result, "[ 'one', 'two', 'three' ]");
+    });
+
+    test(
+        'makeDescription return a String with '
+        'single qouted strings and unqouted other object', () {
+      final values = [
+        'one',
+        'two',
+        'three',
+        true,
+        false,
+        4,
+        5,
+        6,
+        7.0,
+        8.0,
+        9.0,
+      ];
+
+      final result = pTest.makeDescription(baseDescription, 1, values, null);
```

Comment:
What happens if we put a complex object with a custom .tostring in here, is that called for building the description? Can we test that?

Reply:
Added test. This is only sort of a thing when using the custom description builder. Normally the description is passed to the `group` test function which handles this out of the box. Also the custom description builder returns a Object which `group` will resolve.